statefun_tasks/_protobuf.py

```diff
@@ -0,0 +1,179 @@
+from google.protobuf.wrappers_pb2 import DoubleValue, Int64Value, BoolValue, StringValue, BytesValue
+from google.protobuf.any_pb2 import Any
+from google.protobuf.message import Message
+from .messages_pb2 import MapOfStringToAny, ArrayOfAny, TupleOfAny, TaskEntry, GroupEntry, NoneValue, \
+    TaskRetryPolicy, TaskRequest, TaskResult, TaskException, TaskState, GroupResults, Pipeline, PipelineEntry, Address
+from ._utils import _is_tuple
+from typing import Union
+ 
+
+_SCALAR_TYPE_MAP = {
+    float: DoubleValue,
+    int: Int64Value,
+    bool: BoolValue,
+    str: StringValue,
+    bytes: BytesValue
+}
+
+_KNOWN_PROTO_TYPES = [
+    # wrappers
+    DoubleValue,
+    Int64Value,
+    BoolValue,
+    StringValue,
+    BytesValue,
+    NoneValue,
+    
+    # flink task types
+    MapOfStringToAny,
+    TupleOfAny,
+    ArrayOfAny,
+    TaskEntry,
+    GroupEntry,
+    TaskRetryPolicy,
+    TaskRequest,
+    TaskResult,
+    TaskException,
+    TaskState,
+    GroupResults,
+    Pipeline, 
+    PipelineEntry,
+    Address
+]
+
+def _wrap_value(v):
+    # if none return NoneValue wrapper
+    if v is None:
+        return NoneValue()
+
+    python_type = type(v)
+    # wrap scalars in protobuf wrappers
+    if python_type in _SCALAR_TYPE_MAP:
+        mapped = _SCALAR_TYPE_MAP[python_type]()
+        mapped.value = v
+    # leave other protobufs alone
+    elif isinstance(v, Message):
+        mapped = v
+    else:
+        raise ValueError(f'Cannot wrap non-scalar {type(v)} in a protobuf.  Try converting to protobuf first.')
+
+    return mapped
+
+
+def _unwrap_value(v):
+    # if NoneValue wrapper return None
+    if isinstance(v, NoneValue):
+        return None
+
+    proto_type = type(v)
+    # unwrap scalars in protobuf wrappers
+    if proto_type in _SCALAR_TYPE_MAP.values():
+        return v.value
+    return v
+
+
+def _wrap_any(value) -> Any:
+    any = Any()
```

Comment:
I know this is in the protobuf examples, but naming the variables "any" hides the built in any function. Perhaps we should go for something like obj or result instead?

Reply:
done